pkg/auxnetdevice/auxNetDeviceProvider.go

```diff
@@ -0,0 +1,193 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package auxnetdevice
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/jaypipes/ghw"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/utils"
+)
+
+const (
+	maxVendorNameLen  = 20
+	maxProductNameLen = 40
+	classIDBaseInt    = 16
+	classIDBitSize    = 64
+)
+
+type auxNetDeviceProvider struct {
+	deviceList []*ghw.PCIDevice
+	rFactory   types.ResourceFactory
+}
+
+// NewAuxNetDeviceProvider DeviceProvider implementation from auxNetDeviceProvider instance
+func NewAuxNetDeviceProvider(rf types.ResourceFactory) types.DeviceProvider {
+	return &auxNetDeviceProvider{
+		rFactory:   rf,
+		deviceList: make([]*ghw.PCIDevice, 0),
+	}
+}
+
+func (ap *auxNetDeviceProvider) GetDiscoveredDevices() []*ghw.PCIDevice {
+	return ap.deviceList
+}
+
+func (ap *auxNetDeviceProvider) GetDevices(rc *types.ResourceConfig) []types.HostDevice {
+	newAuxDevices := make([]types.HostDevice, 0)
+	for _, device := range ap.deviceList {
+		// discover auxiliary device names
+		auxDevs, err := utils.GetAuxNetDevicesFromPci(device.Address)
+		if err == nil {
+			for _, auxDev := range auxDevs {
+				if newDevice, err := NewAuxNetDevice(device, auxDev, ap.rFactory, rc); err == nil {
+					newAuxDevices = append(newAuxDevices, newDevice)
+				} else {
+					glog.Errorf("auxnetdevice GetDevices(): error creating new device: %q", err)
+				}
+			}
+		} else {
+			glog.Errorf("auxnetdevice GetDevices(): error getting auxnetdevices from device %s: %q",
+				device.Address, err)
+		}
+	}
+	return newAuxDevices
+}
+
+func (ap *auxNetDeviceProvider) AddTargetDevices(devices []*ghw.PCIDevice, deviceCode int) error {
+	for _, device := range devices {
+		devClass, err := strconv.ParseInt(device.Class.ID, classIDBaseInt, classIDBitSize)
+		if err != nil {
+			glog.Warningf("auxnetdevice AddTargetDevices(): unable to parse device class for device %+v %q", device, err)
+			continue
+		}
+
+		if devClass == int64(deviceCode) {
+			vendor := device.Vendor
+			vendorName := vendor.Name
+			if len(vendor.Name) > maxVendorNameLen {
+				vendorName = string([]byte(vendorName)[0:17]) + "..."
+			}
+			product := device.Product
+			productName := product.Name
+			if len(product.Name) > maxProductNameLen {
+				productName = string([]byte(productName)[0:37]) + "..."
+			}
+			glog.Infof("auxnetdevice AddTargetDevices(): device found: %-12s\t%-12s\t%-20s\t%-40s", device.Address,
+				device.Class.ID, vendorName, productName)
+			// exclude netdevice in-use in host XXX ???
```

Comment:
why the ??? are you not sure if to skip netdevice in use?
So for VF this is what we do so I think it fine just need to logf that we as skipping that device because it has default route

Reply:
Ok

Reply:
In case of aux.devs it is parent PCI device, which is not used in the pool.